moveit_ros/perception/pointcloud_octomap_updater/include/moveit/pointcloud_octomap_updater/pointcloud_octomap_updater.h

```diff
@@ -64,8 +66,29 @@ class PointCloudOctomapUpdater : public OccupancyMapUpdater
   void start() override;
   void stop() override;
   ShapeHandle excludeShape(const shapes::ShapeConstPtr& shape) override;
+
+  /** @brief Exclude a shape from the octomap when processing.  Allows specifying an initial transform to put in the
+     transform cache.
+             For use outside of the perception pipeline
+      @param shape The shape to exclude
+      @param pose Pose of the shape in the sensor frame */
+  ShapeHandle excludeShape(const shapes::ShapeConstPtr& shape, const Eigen::Isometry3d& pose);
```

Comment:
I think I just didn't consider this because `Eigen::Isometry3d::Identity()` does not really make sense as a default value - the default value for the `ShapeHandle excludeShape(const shapes::ShapeConstPtr& shape) override;` signature is actually "uninitialized because it will be set in the updateTransformCache callback", and there are very few situations where we would actually want to use the identity pose as the transform.  We also only want to use the version taking an Isometry3d when we are not using the perception pipeline.  So they are basically two separate APIs, even if the contents of the functions happen to be nearly the same.